packages/edit-site/src/components/dataviews/text-filter.js

```diff
@@ -14,12 +14,16 @@ export default function TextFilter( { view, onChangeView } ) {
 	const [ search, setSearch, debouncedSearch ] = useDebouncedInput(
 		view.search
 	);
+	const onChangeViewRef = useRef( onChangeView );
 	useEffect( () => {
-		onChangeView( ( currentView ) => ( {
+		onChangeViewRef.current = onChangeView;
+	}, [ onChangeView ] );
+	useEffect( () => {
+		onChangeViewRef.current( ( currentView ) => ( {
 			...currentView,
 			search: debouncedSearch,
 		} ) );
-	}, [ debouncedSearch, onChangeView ] );
+	}, [ debouncedSearch ] );
```

Comment:
The change here is necessary because it was creating an infinite loop if (the onChangeView depended on the previous view object).

Reply:
Can you explain a bit more when that happened?

Reply:
Sure, the onChangeView now as defined in this PR is a function that depends on the previous "view" value. And the callback here was being called to change the "view" when "onChangeView" changes, so basically it's an infinite loop: view changes onChangeView which triggers this effect that changes view again...